.github/workflows/ci.yml

```diff
@@ -0,0 +1,191 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - master
+    paths-ignore:
+      - 'docs/**'
+      - README.md
+
+  pull_request:
+    paths-ignore:
+      - 'docs/**'
+      - README.md
+
+jobs:
+  frontend:
+    name: Frontend
+    runs-on: ubuntu-16.04
+    env:
+      JOBS: 1 # See https://git.io/vdao3 for details.
+      PERCY_PARALLEL_TOTAL: 2
+      # Percy secrets are included here to enable Percy's GitHub integration
+      # on community-submitted PRs
+      PERCY_TOKEN: 0d8707a02b19aebbec79bb0bf302b8d2fa95edb33169cfe41b084289596670b1
+      PERCY_PROJECT: crates-io/crates.io
+
+    steps:
+      - uses: actions/checkout@v2-beta
+
+      - uses: actions/cache@v1
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
+
+      - uses: actions/setup-node@v1
+        with:
+          node-version: '12.x'
+
+      - name: Install node modules
+        run: npm ci
+
+      - name: Lint
+        run: |
+          npm run lint:hbs
+          npm run lint:js
+          npm run lint:deps
+
+      - name: Test
+        run: npm test
+
+  backend:
+    name: Backend
+    runs-on: ubuntu-16.04
+    strategy:
+      # TODO: [ci] should be true if GitHub Actions supports ""allow failures" on matrix
+      fail-fast: false
+      matrix:
+        rust:
+          - stable
+          - beta
+          - nightly
+
+    env:
+      DATABASE_URL: postgres://postgres:@localhost/cargo_registry_test
+      TEST_DATABASE_URL: postgres://postgres:@localhost/cargo_registry_test
+      CARGO_INCREMENTAL: 0
+      RUSTFLAGS: "-C debuginfo=0 -D warnings"
+
+    steps:
+      - uses: actions/checkout@v2-beta
+
+      - name: Cleanup pre-installed Rust toolchains
+        # The pre-installed toolchain is under root permission, and this would
+        # make tarball fail to extract. Here we remove the symlink and install
+        # our own Rust toolchain if necessary.
+        run: |
+          which rustup
+          which cargo
+          if [[ -L "$HOME/.cargo" && -L "$HOME/.rustup" ]]; then
+              rm -v "$HOME/.rustup"
+              rm -v "$HOME/.cargo"
+          fi
+          echo "::add-path::$HOME/.cargo/bin"
+
+      - name: Cache rustup
+        uses: actions/cache@v1
+        with:
+          path: ~/.rustup
+          key: ${{ runner.os }}-rustup-${{ matrix.rust }}-${{ hashFiles('**/Cargo.lock') }}
```

Comment:
I'm not sure if it is necessary to include the `-${{ hashFiles('**/Cargo.lock') }}` portion for this directory and for the `~/.cargo/` paths below.  (It definitely makes sense for the `target/` directory.)  My reasoning is that the contents of these directories shouldn't really be affected by the contents of the lock file.

By the same reasoning, `-${{ matrix.rust }}` may also not be necessary for many of these.

If you're unsure about removing it though, I definitely don't see this as a blocker.  This is probably a micro-optimization given the expected size of these directories.

Reply:
Thanks. We need figure out when to invalidate old cache. Maybe some trick like [script//ci/cargo-clean-on-new-rustc-version.sh](https://github.com/rust-lang/crates.io/blob/710ad1edc03c5ac3542c16821232ef6b365f098a/script/ci/cargo-clean-on-new-rustc-version.sh), but I am not sure how to implement that.